src/macros.rs

```diff
@@ -149,12 +149,12 @@ pub fn rewrite_macro(
     shape: Shape,
     position: MacroPosition,
 ) -> Option<String> {
-    context.inside_macro.replace(true);
+    let old_value = context.inside_macro.replace(true);
```

Comment:
Could you use an RAII pattern here so there is no change of missing the `replace` at the end of the function?

Reply:
Hmm, I am sorry, but I am not sure how to apply an RAII pattern here. Does `RefCell` provides such thing? Either way, I have tried to mimic the pattern here. Please let me know if this is not what you expected, or there is a better way to do this!

Reply:
RAII in Rust would be something like:

```
struct InsideMacroRestore<'a> {
    context: &RewriteContext,
    old_value: bool,
}

impl<'a> InsideMacroRestore<'a> {
    // Call this as `let _imr = InsideMacroRestore(context, true);` which sets
    // the value and stores the old one.
    fn replace(context: &'a RewriteContext, new_value: bool) -> InsideMacroRestore<'a> {
        let old_value = context.inside_macro.replace(new_value);
        InsideMacroRestore { context, old_value }
    }
}

impl<'a> Drop for InsideMacroRestore<'a> {
    // Called automatically when `_imr` goes out of scope, which restores the old
    // value of inside_macro
    fn drop(&mut self) {
        self.context.inside_macro.replace(self.old_value)
    }
}
```

`MutexLock` is an example of such thing, although there the object is actually useful, here we just want the `Drop` impl